PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/RuleConfigLoader.java

```diff
@@ -221,4 +225,32 @@ private Map<String, String> getMapOfDiscouragedIdentifiers(JsonReader reader, Ma
         }
         return discouragedIdentifiersToCheckMap;
     }
+
+    /**
+     * This method parses the values from the JSON object
+     *
+     * @param reader - the JsonReader object to read the JSON file
+     * @return List of strings parsed from the JSON object
+     * @throws IOException - if there is an error reading the file
+     */
+    private List<String> getValuesFromJsonReader(JsonReader reader) throws IOException {
+
+        List<String> values = new ArrayList<>();
+
+        // Check if the JSON file starts with an object
+        if (reader.nextToken() != JsonToken.START_OBJECT) {
+            throw new IOException("Expected start of object");
```

Comment:
I am a fan of using unchecked exceptions.  This wouldn't' be the right exception to throw though because it's not an IO problem.  https://azure.github.io/azure-sdk/java_introduction.html#exceptions

Reply:
used IllegalArgumentException with a specific error messages